src/app/page.tsx

```diff
@@ -1,36 +1,35 @@
 import Link from "next/link";
 
-import { LatestPost } from "~/app/_components/post";
-import { getServerAuthSession } from "~/server/auth";
 import { api, HydrateClient } from "~/trpc/server";
-import { UserProfile } from "./_components/userProfile/UserProfile";
+import { getServerAuthSession } from "~/server/auth";
+import UserProfile from "~/app/_components/userProfile/UserProfile";
 
 export default async function Home() {
-  const hello = await api.post.hello({ text: "from tRPC" });
-  const session = await getServerAuthSession();
-
-  void api.post.getLatest.prefetch();
+  const SESSION = await getServerAuthSession();
```

Comment:
Also, let's do this with `getServerSideProps()`. Refer to this [documentation](https://create.t3.gg/en/usage/next-auth#retrieving-session-server-side)

Reply:
Thanks for the review, but I am don't quite understand why we would want to do that. We are already fetching the session on the server